core/src/Platforms/android/TokenCacheAccessor.cs

```diff
@@ -70,43 +71,52 @@ public TokenCacheAccessor(RequestContext requestContext) : this()
             _requestContext = requestContext;
         }
 
-        public void SaveAccessToken(string cacheKey, string item)
+        public void SaveAccessToken(MsalAccessTokenCacheItem item)
         {
             ISharedPreferencesEditor editor = _accessTokenSharedPreference.Edit();
-            editor.PutString(cacheKey, item);
+            editor.PutString(item.GetKey().ToString(), JsonHelper.SerializeToJson(item));
             editor.Apply();
         }
 
-        public void SaveRefreshToken(string cacheKey, string item)
+        public void SaveRefreshToken(MsalRefreshTokenCacheItem item)
         {
             ISharedPreferencesEditor editor = _refreshTokenSharedPreference.Edit();
-            editor.PutString(cacheKey, item);
+            editor.PutString(item.GetKey().ToString(), JsonHelper.SerializeToJson(item));
             editor.Apply();
         }
 
-        public string GetRefreshToken(string refreshTokenKey)
+        public void SaveIdToken(MsalIdTokenCacheItem item)
         {
-            return _refreshTokenSharedPreference.GetString(refreshTokenKey, null);
+            ISharedPreferencesEditor editor = _idTokenSharedPreference.Edit();
```

Comment:
Too much duplicated code...

Reply:
see related comments. pls. take in the code style discussion

---
In reply to: [199549788](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1108#discussion_r199549788) [](ancestors = 199549788)